.github/actions/set-container-repo-and-determine-image-tag/action.yml

```diff
@@ -58,3 +64,30 @@ runs:
         fi
 
         echo "image-tag=${image_tag}" >> $GITHUB_OUTPUT
+    - name: Output gadgets repository
+      id: gadgets-repo
+      shell: bash
+      run: |
+        if [ -z '${{ inputs.registry }}' ]; then
+          echo 'inputs.registry should not be empty! Set it before calling this action' 1>&2
+          exit 1
+        fi
+
+        gadgets_repo=${{ inputs.registry }}/${{ github.repository_owner }}
+        if [ ${{ github.event_name }} == 'pull_request' ]; then
+          gadgets_repo=ttl.sh/${{ github.sha }}
```

Comment:
This is the solution I came up with, we could change it to share the gadget image as artifacts between the different jobs, but I like it because it's also testing the push / pull functionality.

What concerns do you have with ttl.sh?

Reply:
I don't think it's too bad. We're not using it to host our production images, and there is not an API we use, just push and pulling containers. I'll keep it for now, later on we can try a different approach.